### PR TITLE
Adding debug information when connection fails and enhancements to ex…

### PR DESCRIPTION
**Pull request by s-shrestha**
Adding debug information when connection fails and enhancements to execution timer

### CODE REVIEW

File: src/include/baselib/core/Utils.h [hunk 1]
```diff
@@ -273,17 +273,26 @@ namespace bl
 
             std::string                                             m_name;
             Logging::Channel*                                       m_channel;
+            long                                                    m_durationThresholdInSeconds;
             decltype( time::microsec_clock::universal_time() )      m_startTime;
+            bl::cpp::ScalarTypeIniter< bool >                       m_canceled;
```
Comment (lazar-ivanov) [inline src/include/baselib/core/Utils.h]: unnecessary bl:: prefix

File: src/include/baselib/core/Utils.h [hunk 2]
```diff
@@ -300,23 +310,39 @@ namespace bl
...
-                BL_LOG(
-                    *m_channel,
-                    BL_MSG()
-                        << ( m_name.empty() ? "Command" : m_name )
-                        << " completed in "
-                        << time::to_simple_string( duration )
-                    );
+                    const auto durationThresholdInSecondsSatisfied =
+                        m_durationThresholdInSeconds == NO_DURATION_THRESHOLD ||
+                        duration > bl::time::time_duration( bl::time::seconds( m_durationThresholdInSeconds ) );
```
Comment (lazar-ivanov) [inline src/include/baselib/core/Utils.h]: unnecessary bl:: prefix (two spots)

Comment (s-shrestha) [reply src/include/baselib/core/Utils.h]: Will fix.

File: src/include/baselib/core/Utils.h (hunk 1 above)
Comment (s-shrestha) [reply src/include/baselib/core/Utils.h]: Will fix.

**Pull request merged**